matrix-matrix-product.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * @brief Performs matrix-matrix multiplication.
+ *
+ * This function multiplies two matrices `m1` and `m2` and returns the resultant matrix.
+ * It assumes that the dimensions of the matrices are compatible for multiplication,
+ * i.e., the number of columns in `m1` equals the number of rows in `m2`.
+ *
+ * @param m1 The first input matrix (a x b).
+ * @param m2 The second input matrix (b x c).
+ * @return The resultant matrix after multiplication (a x c).
+ *
+ * @throws std::exit(1) if either matrix does not have more than one column.
+ */
+#include <vector>
+#include <iostream>
+
+using std::vector;
+typedef vector<vector<double>> Matrix;
+
+Matrix matrix_matrix_product(const Matrix& m1, const Matrix& m2){
+	int a = m1.size(); // number of rows of m1
+	int b = m1[0].size(); // number of columns of m1
```

Comment:
One subtle issue is that your datastructure supports every row being a different length. which could potentially cause issues.